### PR TITLE
Feature: support configuration of the input power threshold for overscaling per inverter

### PR DESCRIPTION
**Pull request by AndreasBoehm**
Based on @leoai-81's work that can be found here: https://github.com/hoylabs/OpenDTU-OnBattery/pull/1298

Screenshot 2024-12-23 at 00 10 21

### CODE REVIEW

File: include/Configuration.h [hunk 1]
```diff
@@ -141,6 +141,7 @@ struct POWERLIMITER_INVERTER_CONFIG_T {
     bool UseOverscalingToCompensateShading;
     uint16_t LowerPowerLimit;
     uint16_t UpperPowerLimit;
+    int32_t ScalingThreshold;
```
Comment (schlimmchen) [inline include/Configuration.h:144]: Why is this a signed int? I think uint8_t is better suited, is it not? This is 0 - 100.

File: src/PowerLimiterSolarInverter.cpp [hunk 2]
```diff
@@ -147,8 +147,8 @@ uint16_t PowerLimiterSolarInverter::scaleLimit(uint16_t expectedOutputWatts)
         // is currently not producing (efficiency is zero in that case)
         inverterEfficiencyFactor = (inverterEfficiencyFactor > 0) ? inverterEfficiencyFactor/100 : 0.967;
 
-        // 98% of the expected power is good enough
-        auto expectedAcPowerPerMppt = (getCurrentLimitWatts() / dcTotalMppts) * 0.98;
+        auto scalingThreshold = static_cast<float>(_config.ScalingThreshold) / 100.0;
+        auto expectedAcPowerPerMppt = (getCurrentLimitWatts() / dcTotalChnls) * scalingThreshold;
```
Comment (schlimmchen) [inline src/PowerLimiterSolarInverter.cpp:151]: You changed from dcMppts to dcTotalChnls. I don't see why. Is this by accident? Is it effectively the same?

File: webapp/src/views/PowerLimiterAdminView.vue [hunk 3]
```diff
@@ -137,6 +137,19 @@
                             wide
                         />
 
+                        <InputElement
+                            v-if="powerLimiterConfigList.inverters[idx].use_overscaling_to_compensate_shading"
+                            :label="$t('powerlimiteradmin.ScalingPowerThreshold')"
+                            v-model="powerLimiterConfigList.inverters[idx].scaling_threshold"
+                            :tooltip="$t('powerlimiteradmin.ScalingPowerThresholdHint')"
+                            placeholder="98"
```
Comment (schlimmchen) [inline webapp/src/views/PowerLimiterAdminView.vue:145]: Let's not use placeholders. This has confused people in the past, including me. The placeholder should not be needed at all as you initialize the JSON value to the #define, so there is always a value in input field anyways.

**Review by schlimmchen — CHANGES_REQUESTED** [review]

File: include/Configuration.h (hunk 1 above)
Comment (AndreasBoehm) [reply include/Configuration.h:144]: Copy&paste mistake as i took most of the code from leoai-82

File: src/PowerLimiterSolarInverter.cpp (hunk 2 above)
Comment (AndreasBoehm) [reply src/PowerLimiterSolarInverter.cpp:151]: Copy&paste mistake as i took most of the code from leoai-82

File: webapp/src/views/PowerLimiterAdminView.vue (hunk 3 above)
Comment (AndreasBoehm) [reply webapp/src/views/PowerLimiterAdminView.vue:145]: Makes sense

File: src/PowerLimiterSolarInverter.cpp (hunk 2 above)
Comment (schlimmchen) [reply src/PowerLimiterSolarInverter.cpp]: Oh... Okay, tell me next time, so I will be more thorough. I was close to letting this slip through as I assumed you know what you wre doking :speak_no_evil: :wink:

**Pull request merged**